From 46b90d629bad9ede16de084675d44fd126aaadd5 Mon Sep 17 00:00:00 2001
From: Neal Liu <neal_liu@aspeedtech.com>
Date: Thu, 18 Sep 2025 10:43:45 +0800
Subject: [PATCH 30/32] arch: arm: mach-aspeed: add secure boot support

Set env verify=yes if secure boot is enabled to force
FIT image signature verification in AST2700.

Signed-off-by: Neal Liu <neal_liu@aspeedtech.com>
Change-Id: Ic3dc96f8ab828d623ba34c3c86cba9860a35271a
(cherry picked from commit 7e5ec2150b1aa3e128d5cad8b499bcc5b12505d0)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/include/asm/arch-aspeed/scu_ast2700.h | 1 +
 arch/arm/mach-aspeed/ast2700/platform.c        | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm/include/asm/arch-aspeed/scu_ast2700.h b/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
index 1f0b4d06d0..7f496a3bbc 100644
--- a/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
+++ b/arch/arm/include/asm/arch-aspeed/scu_ast2700.h
@@ -70,6 +70,7 @@
 /* SoC1 SCU Register */
 #define SCU_IO_HWSTRAP_UFS			BIT(23)
 #define SCU_IO_HWSTRAP_EMMC			BIT(11)
+#define SCU_IO_HWSTRAP_SECBOOT			BIT(5)
 #define SCU_IO_HWSTRAP_LTPI0_EN			BIT(3)
 #define SCU_IO_HWSTRAP_LTPI1_EN			BIT(1)
 
diff --git a/arch/arm/mach-aspeed/ast2700/platform.c b/arch/arm/mach-aspeed/ast2700/platform.c
index 9a460ee705..abf70b5ede 100644
--- a/arch/arm/mach-aspeed/ast2700/platform.c
+++ b/arch/arm/mach-aspeed/ast2700/platform.c
@@ -48,6 +48,11 @@ int arch_misc_init(void)
 			env_set("boot_device", "spi");
 			spi_bootarg_config();
 		}
+
+		if ((readl(ASPEED_IO_HW_STRAP1) & SCU_IO_HWSTRAP_SECBOOT))
+			env_set("verify", "yes");
+		else
+			env_set("verify", "no");
 	}
 
 	return 0;
-- 
2.34.1

